### ISSUE TITLE
Segfault for YOLOv11 inference when Gstreamer is imported before ultralytics

### ISSUE THREAD

**Issue by Toufengmehmet09** [ultralytics/ultralytics#21486]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Predict

### Bug

When i import and initialize gstreamer library before ultralytics library in python i get strange segfaults, when trying to do YOLOv11 inference.

Error-message:
```
0: 384x480 (no detections), 27.8ms
Speed: 2.4ms preprocess, 27.8ms inference, 14.2ms postprocess per image at shape (1, 3, 384, 480)
Fatal Python error: Aborted

Thread 0x00007ff65d5a8640 (most recent call first):
  File "/usr/lib/python3.10/socket.py", line 955 in getaddrinfo
  File "/home/user/project/venv/lib/python3.10/site-packages/urllib3/util/connection.py", line 60 in create_connection
  File "/home/user/project/venv/lib/python3.10/site-packages/urllib3/connection.py", line 198 in _new_conn
  File "/home/user/project/venv/lib/python3.10/site-packages/urllib3/connection.py", line 753 in connect
  File "/home/user/project/venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 1093 in _validate_conn
  File "/home/user/project/venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 464 in _make_request
  File "/home/user/project/venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 787 in urlopen
  File "/home/user/project/venv/lib/python3.10/site-packages/requests/adapters.py", line 667 in send
  File "/home/user/project/venv/lib/python3.10/site-packages/requests/sessions.py", line 703 in send
  File "/home/user/project/venv/lib/python3.10/site-packages/requests/sessions.py", line 589 in request
  File "/home/user/project/venv/lib/python3.10/site-packages/requests/api.py", line 59 in request
  File "/home/user/project/venv/lib/python3.10/site-packages/ultralytics/hub/utils.py", line 100 in requests_with_progress
  File "/home/user/project/venv/lib/python3.10/site-packages/ultralytics/hub/utils.py", line 152 in func
  File "/usr/lib/python3.10/contextlib.py", line 79 in inner
  File "/usr/lib/python3.10/threading.py", line 953 in run
  File "/usr/lib/python3.10/threading.py", line 1016 in _bootstrap_inner
  File "/usr/lib/python3.10/threading.py", line 973 in _bootstrap

Current thread 0x00007ff7f5db3000 (most recent call first):
  File "/home/user/project/venv/lib/python3.10/site-packages/ultralytics/engine/results.py", line 187 in __getitem__
  File "/home/user/Desktop/PROJECTS/multitracker/mre.py", line 16 in <module>

Extension modules: numpy.core._multiarray_umath, numpy.core._multiarray_tests, numpy.linalg._umath_linalg, numpy.fft._pocketfft_internal, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, PIL._imaging, kiwisolver._cext, pandas._libs.tslibs.ccalendar, pandas._libs.tslibs.np_datetime, pandas._libs.tslibs.dtypes, pandas._libs.tslibs.base, pandas._libs.tslibs.nattype, pandas._libs.tslibs.timezones, pandas._libs.tslibs.fields, pandas._libs.tslibs.timedeltas, pandas._libs.tslibs.tzconversion, pandas._libs.tslibs.timestamps, pandas._libs.properties, pandas._libs.tslibs.offsets, pandas._libs.tslibs.strptime, pandas._libs.tslibs.parsing, pandas._libs.tslibs.conversion, pandas._libs.tslibs.period, pandas._libs.tslibs.vectorized, pandas._libs.ops_dispatch, pandas._libs.missing, pandas._libs.hashtable, pandas._libs.algos, pandas._libs.interval, pandas._libs.lib, pandas._libs.ops, pandas._libs.hashing, pandas._libs.arrays, pandas._libs.tslib, pandas._libs.sparse, pandas._libs.internals, pandas._libs.indexing, pandas._libs.index, pandas._libs.writers, pandas._libs.join, pandas._libs.window.aggregations, pandas._libs.window.indexers, pandas._libs.reshape, pandas._libs.groupby, pandas._libs.json, pandas._libs.parsers, pandas._libs.testing, gi._gi, cairo._cairo, gi._gi_cairo, torch._C, torch._C._dynamo.autograd_compiler, torch._C._dynamo.eval_frame, torch._C._dynamo.guards, torch._C._dynamo.utils, torch._C._fft, torch._C._linalg, torch._C._nested, torch._C._nn, torch._C._sparse, torch._C._special, yaml._yaml, PIL._imagingft, _brotli, simplejson._speedups, charset_normalizer.md (total: 74)

Process finished with exit code 134 (interrupted by signal 6:SIGABRT)
```

### Environment

Ultralytics 8.3.168 🚀 Python-3.10.12 torch-2.7.1+cu126 CUDA:0 (NVIDIA GeForce RTX 4090 Laptop GPU, 15930MiB)
Setup complete ✅ (32 CPUs, 62.6 GB RAM, 1172.1/1374.0 GB disk)

OS                  Linux-6.8.0-64-generic-x86_64-with-glibc2.35
Environment         Linux
Python              3.10.12
Install             git
Path                /home/user/project/venv/lib/python3.10/site-packages/ultralytics
RAM                 62.60 GB
Disk                1172.1/1374.0 GB
CPU                 13th Gen Intel Core(TM) i9-13900HX
CPU count           32
GPU                 NVIDIA GeForce RTX 4090 Laptop GPU, 15930MiB
GPU count           1
CUDA                12.6

numpy               ✅ 1.25.1>=1.23.0
matplotlib          ✅ 3.10.3>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 11.3.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.4>=2.23.0
scipy               ✅ 1.15.3>=1.4.1
torch               ✅ 2.7.1>=1.8.0
torch               ✅ 2.7.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.22.1>=0.9.0
tqdm                ✅ 4.67.1>=4.64.0
psutil              ✅ 7.0.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.3.1>=1.1.4
ultralytics-thop    ✅ 2.0.14>=2.0.0

### Minimal Reproducible Example

```
import faulthandler
faulthandler.enable()
import gi
gi.require_version('Gst', '1.0')
from gi.repository import Gst
Gst.init(None)

from ultralytics import YOLO
import numpy as np

model = YOLO("yolo11n.pt", task="detect")
img = np.zeros((512, 640, 3), dtype=np.uint8)
results = model(img, imgsz=(640, 480), verbose=True)
for result in results:
    boxes = result.boxes
    for box in boxes:
        xyxy = box.xyxy.cpu().numpy()[0]
        print(xyxy)
```

### Additional

GStreamer 1.20.3
PyGObject 3.42.1

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Toufengmehmet09, thank you for reporting this issue and for providing detailed information and a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/)—this is super helpful! 🚀

This is an automated response to let you know that an Ultralytics engineer will review and assist you soon.

In the meantime, please make sure you are using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) with [**Python>=3.8**](https://www.python.org/) and [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). You can upgrade with:

```bash
pip install -U ultralytics
```

For new users, the [Docs](https://docs.ultralytics.com/) are a great resource, including many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples.

If you wish to experiment or debug in different environments, YOLO can be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Join the Ultralytics community where it suits you best! For real-time chat, check out [Discord](https://discord.com/invite/ultralytics) 🎧, for in-depth discussions visit [Discourse](https://community.ultralytics.com/), or head to our [Subreddit](https://www.reddit.com/r/Ultralytics/) to connect with others.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If the badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you again for your report! An Ultralytics engineer will follow up with you soon 🔎

**Comment by Y-T-G** [comment]
I don't think this is Ultralytics related. You can try upgrading your Python version.

**Comment by nikunjlad** [comment]
@Toufengmehmet09 I tried to recreate your test case here. I have a 3060TI Nvidia GPU with x86 system running Ubuntu 22.04. I created a virtualenv and essentially installed all your libraries above. I have CUDA 12.2 vs your 12.6. I don't think that should be much of an issue here. I have the same PyGObject and ultralytics version as you. I am able to successfully do inference and get below result:

<img width="943" height="267" alt="Image" src="https://github.com/user-attachments/assets/ec41a3c2-bb23-4528-9ae7-0cea3e68baf7" />

I feel like the issue is with your network and probably because it could not fully download the weights. I would suggest you to try again. In case you are curious, below is my  package list in my virtualenv if you want to try it out. I used Python 3.10 as well. 

[requirements.txt](https://github.com/user-attachments/files/21480292/requirements.txt)

**Comment by Toufengmehmet09** [comment]
@nikunjlad I have verified the neural network to work. I used both the downloaded official weights as well as self-trained ones. However i found that it is a workaround to import ultralytics first and PyGObject second. That prevents the issue from happening. If you have any other ideas what i should check please tell me.

**Comment by nikunjlad** [comment]
@Toufengmehmet09 So you are still facing issues importing PyGObject first?

**Comment by Toufengmehmet09** [comment]
@nikunjlad That is right, i went for a workaround be importing ultralytics first. But whenever i import PyGObject first the tensors received by the gpu will behave differently. They produce valid results, however iterating over them or using other build-in functions can cause a segfault. However they contain valid yolo results.

**Comment by nikunjlad** [comment]
@Toufengmehmet09 can you try creating a new Python 3.10 virtualenv and installing the pip packages using my requirements.txt file I attached in the earlier post above? Try to see if it stills errors out. I can think of 2 things-
1. Library versions mismatch causing compatibility issues
2. Some shared library is not in your system path causing you to get segmentation fault

**Comment by Toufengmehmet09** [comment]
@nikunjlad I used your requirements.txt file and the issue is gone in a fresh venv environment. Must be because of incompatible packages. Thanks
